Rakefile

```diff
@@ -4,7 +4,24 @@ require "rubocop/rake_task"
 RSpec::Core::RakeTask.new(:spec)
 RuboCop::RakeTask.new
 
-task default: ["spec", "rubocop:auto_correct"]
+desc "Validate all files match their respective schemas"
+task schema: ["schema:links"]
+
+namespace :schema do
+  desc "Validate links are well-formed"
+  task :links do
+    require_relative "lib/links"
+    links_csv = File.expand_path("data/links.csv", __dir__)
+    links = Links.from_csv(links_csv)
+
+    invalid_links = links.all_links.reject(&:valid?)
+    unless invalid_links.empty?
+      raise "Not all of the links were well-formed according to their schema: #{invalid_links.map(&:attributes).join("\n- ")}"
```

Comment:
Although a handy way of doing this, I'm not sure this is super readable to not-as-regular Rubyists - refactor? :wrench:

Reply:
It looks reasonable to me, you could bind the `invalid_links.map` expression to a variable with a meaningful name but I'm not too bothered by it